Plugins/Feedback/NWScript/nwnx_feedback.nss

```diff
@@ -44,15 +74,92 @@ int NWNX_Feedback_GetFeedbackMessageHidden(int nMessage, object oPC = OBJECT_INV
+    NWNX_PushArgumentInt(NWNX_Feedback, sFunc, nMessage);
+    NWNX_PushArgumentInt(NWNX_Feedback, sFunc, nMessageType);
+    NWNX_PushArgumentObject(NWNX_Feedback, sFunc, oPC);
+    NWNX_CallFunction(NWNX_Feedback, sFunc);
+}
+
+/*
+// CombatLog Messages
+// For use with NWNX_Feedback_SetCombatLogMessageHidden() and
+//              NWNX_Feedback_GetCombatLogMessageHidden()
+
+const int COMBATLOG_SIMPLE_ADJECTIVE    = 1;
+const int COMBATLOG_SIMPLE_DAMAGE       = 2;
+const int COMBATLOG_COMPLEX_DAMAGE      = 3;
+const int COMBATLOG_COMPLEX_DEATH       = 4;
+const int COMBATLOG_COMPLEX_ATTACK      = 5;
+const int COMBATLOG_SPECIAL_ATTACK      = 6;
+const int COMBATLOG_SAVING_THROW        = 7;
+const int COMBATLOG_CAST_SPELL          = 8;
+const int COMBATLOG_USE_SKILL           = 9;
+const int COMBATLOG_SPELL_RESISTANCE    = 10;
+const int COMBATLOG_FEEDBACK            = 11; // NOTE: This hides ALL feedback messages, to hide individual messages use NWNX_Feedback_SetFeedbackMessageHidden()
+const int COMBATLOG_COUNTERSPELL        = 12;
+const int COMBATLOG_TOUCHATTACK         = 13;
+const int COMBATLOG_INITIATIVE          = 14;
+const int COMBATLOG_DISPEL_MAGIC        = 15;
+const int COMBATLOG_POLYMORPH           = 17;
+const int COMBATLOG_FEEDBACKSTRING      = 18;
+const int COMBATLOG_VIBRATE             = 19;
+const int COMBATLOG_UNLOCKACHIEVEMENT   = 20;
+
+// 1  -> Simple_Adjective: <charname> : <adjective described by strref>
+// 2  -> Simple_Damage: <charname> damaged : <amount>
+// 3  -> Complex_Damage: <charname> damages <charname> : <amount>
+// 4  -> Complex_Death: <charname> killed <charname>
+// 5  -> Complex_Attack: <charname> attacks <charname> : *hit* / *miss* / *parried* : (<attack roll> + <attack mod> = <modified total>)
+// 6  -> Special_Attack: <charname> attempts <special attack> on <charname> : *success* / *failure* : (<attack roll> + <attack mod> = <modified roll>)
+// 7  -> Saving_Throw: <charname> : <saving throw type> : *success* / *failure* : (<saving throw roll> + <saving throw modifier> = <modified total>)
+// 8  -> Cast_Spell: <charname> casts <spell name> : Spellcraft check *failure* / *success*
+// 9  -> Use_Skill: <charname> : <skill name> : *success* / *failure* : (<skill roll> + <skill modifier> = <modified total> vs <DC> )
+// 10 -> Spell_Resistance: <charname> : Spell Resistance <SR value> : *success* / *failure*
+// 11 -> Feedback: Reason skill/feat/ability failed.
+// 12 -> Counterspel: <charname> casts <spell name> : *spell countered by* : <charname> casting <spell name>
+// 13 -> TouchAttack: <charname> attempts <melee/ranged touch attack> on <charname> : *hit/miss/critical* : (<attack roll> + <attack mod> = <modified roll>)
+// 14 -> Initiative: <charname> : Initiative Roll : <total> : (<roll> + <modifier> = <total>)
+// 15 -> Dispel_Magic: Dispel Magic : <charname> : <spell name>, <spell name>, <spell name>...
+// 17 -> Unused, probably
+// 18 -> Same as 11, probably, maybe
```

Comment:
This is a generic feedback message, which is sent by the server. I think this might even trigger for `SendMessageToPC()`

Reply:
SendMessageToPC(), SendMessageToPCByStrRef(), FloatingTextStringOnCreature() and FloatingTextStrRefOnCreature() all trigger COMBATLOG_FEEDBACK = 11, so I dunno what 18 is for